plugins/woocommerce/src/Internal/DownloadPermissionsAdjuster.php

```diff
@@ -99,7 +99,21 @@ public function adjust_download_permissions( int $product_id ) {
 
 		$children_with_downloads = array();
 		foreach ( $children_ids as $child_id ) {
-			$child                                = wc_get_product( $child_id );
+			$child = wc_get_product( $child_id );
+
+			// Ensure we have a valid child product.
+			if ( ! $child ) {
+				wc_get_logger()->warning(
+					sprintf(
+						/* translators: 1: child product ID 2: parent product ID. */
+						__( 'Unable to load child product %1$d while adjusting download permissions for product %2$d.', 'woocommerce' ),
```

Comment:
Should we call it a "product variation" instead of a "child product" for consistency with the UI?

Reply:
Not sure; I tend to think this is probably the most accurate verbiage. Besides being aligned with the programmatic API—`$product->get_children()`—this code may also run in relation to grouped products, not just variable products (though I realize that isn't the scenario the docblock highlights).

Reply:
Ok, that makes sense.